test/e2e/case12_ordering_test.go

```diff
@@ -0,0 +1,385 @@
+		opt := metav1.ListOptions{}
+		utils.ListWithTimeout(clientHubDynamic, gvrPolicy, opt, 0, true, defaultTimeoutSeconds)
+		utils.ListWithTimeout(clientManagedDynamic, gvrPolicy, opt, 0, true, defaultTimeoutSeconds)
+	})
+	It("Should set to compliant when dep status is compliant", func() {
+		By("Generating an compliant event on the dependency")
+		managedPlc := utils.GetWithTimeout(
+			clientManagedDynamic,
+			gvrPolicy,
+			case12DepName,
+			clusterNamespace,
+			true,
+			defaultTimeoutSeconds)
+		Expect(managedPlc).NotTo(BeNil())
+		managedRecorder.Event(
+			managedPlc,
+			"Normal",
+			"policy: managed/namespace-foo-setup-configpolicy",
+			"Compliant; No violation detected")
+		By("Checking if dependency status is compliant")
+		Eventually(func() interface{} {
+			managedPlc = utils.GetWithTimeout(
+				clientManagedDynamic,
+				gvrPolicy,
+				case12DepName,
+				clusterNamespace,
+				true,
+				defaultTimeoutSeconds)
+
+			return getCompliant(managedPlc)
+		}, defaultTimeoutSeconds, 1).Should(Equal("Compliant"))
+		By("Checking if hub policy status is in sync")
+		Eventually(func() interface{} {
+			hubPlc := utils.GetWithTimeout(
+				clientHubDynamic,
+				gvrPolicy,
+				case12DepName,
+				clusterNamespaceOnHub,
+				true,
+				defaultTimeoutSeconds)
+
+			return hubPlc.Object["status"]
+		}, defaultTimeoutSeconds, 1).Should(Equal(managedPlc.Object["status"]))
+		By("Generating an compliant event on the policy")
+		managedPlc = utils.GetWithTimeout(
+			clientManagedDynamic,
+			gvrPolicy,
+			case12PolicyName,
+			clusterNamespace,
+			true,
+			defaultTimeoutSeconds)
+		Expect(managedPlc).NotTo(BeNil())
+		managedRecorder.Event(
+			managedPlc,
+			"Normal",
+			"policy: managed/case12-config-policy",
+			"Compliant; No violation detected")
+		By("Checking if policy status is compliant")
+		Eventually(func() interface{} {
+			managedPlc = utils.GetWithTimeout(
+				clientManagedDynamic,
+				gvrPolicy,
+				case12PolicyName,
+				clusterNamespace,
+				true,
+				defaultTimeoutSeconds)
+
+			return getCompliant(managedPlc)
+		}, defaultTimeoutSeconds, 1).Should(Equal("Compliant"))
+	})
+	It("Should set to Pending when dep status is NonCompliant", func() {
+		By("Generating a noncompliant event on the policy")
+		managedPlc := utils.GetWithTimeout(
+			clientManagedDynamic,
+			gvrPolicy,
+			case12DepName,
+			clusterNamespace,
+			true,
+			defaultTimeoutSeconds)
+		Expect(managedPlc).NotTo(BeNil())
+		managedRecorder.Event(
+			managedPlc,
+			"Warning",
+			"policy: managed/namespace-foo-setup-configpolicy",
+			"NonCompliant; there is violation")
+
+		By("Checking if dependency status is noncompliant")
+		Eventually(checkCompliance(case12DepName), defaultTimeoutSeconds, 1).Should(Equal("NonCompliant"))
+
+		By("Checking if policy status is pending")
+		Eventually(checkCompliance(case12PolicyName), defaultTimeoutSeconds, 1).Should(Equal("Pending"))
+	})
+	It("Should set to Compliant when dep status is resolved", func() {
```

Comment:
Unnecessary suggestion: This test begins by repeating what the previous test did. If the `Ordered` keyword was added to this parent `Describe`, and the `After/BeforeEach` were converted to `After/BeforeAll`, then the repetition would not be needed.

But that might require other test re-organization, so it's not necessary.

Reply:
I removed the before/afterAll because other tests don't use the same policies, so I think this isn't needed anymore